### PR TITLE
Creation of users via API

### PR DESCRIPTION
**Pull request by Nek-**
This PR adds the following features:

* Doctrine mapping inside « domain » folder. That means you can create `Project\Entity\YourEntity` and call it everywhere with `Project:YourEntity`
* User registration via API

### CODE REVIEW

File: back/src/CiTron/DependencyInjection/Compiler/DoctrineMappingCompilerPass.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                $namespace = 'CiTron\\' . str_replace('/', '\\', $namespace[count($namespace)-1]);
+
+                $mapping[$alias] = $namespace;
+                $folders[$namespace] = $folder;
+            }
+        }
+
+        // Setting the mapping alias -> namespace
+        foreach($mapping as $alias => $namespace) {
+            var_dump($alias);
```
Comment (Nek-) [inline back/src/CiTron/DependencyInjection/Compiler/DoctrineMappingCompilerPass.php]: var_dump :>

File: back/src/CiTron/DependencyInjection/Compiler/DoctrineMappingCompilerPass.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        $chainDriverDef = $container->getDefinition('doctrine.orm.default_metadata_driver');
+
+        foreach (array_keys($folders) as $namespace) {
+            $chainDriverDef->addMethodCall('addDriver', [
+                new Reference('doctrine.orm.default_annotation_metadata_driver'),
+                $namespace,
+            ]);
+        }
+
+        // Adding folders to driver
```
Comment (Nek-) [inline back/src/CiTron/DependencyInjection/Compiler/DoctrineMappingCompilerPass.php]: hum ?

File: back/app/config/routing/users.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+user_registration:
+    path: /registration
+    defaults: { _controller: 'CiTron:User\Registration:createUser' }
```
Comment (valanz) [inline back/app/config/routing/users.yml]: Shouldn't we authorize only `POST` request?

File: back/src/CiTron/Controller/User/RegistrationController.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+use CiTron\Controller\Controller;
+use CiTron\Symfony\HttpFoundation\JsonResponse;
+use CiTron\User\Entity\User;
+use CiTron\User\Form\UserType;
+use Symfony\Component\HttpFoundation\Request;
+
+class RegistrationController extends Controller
+{
+    public function createUserAction(Request $request)
```
Comment (valanz) [inline back/src/CiTron/Controller/User/RegistrationController.php]: Missing docs

File: back/src/CiTron/User/Entity/User.php [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+     * @ORM\Column(name="username", type="string", length=255, unique=true)
+     */
+    private $username;
+
+    /**
+     * It should be clean while the user is retrieved
+     * @var string
+     *
+     * @ORM\Column(name="password", type="string")
+     * @Assert\Length(min = 4, minMessage="Password too short")
```
Comment (valanz) [inline back/src/CiTron/User/Entity/User.php]: We should think of a stronger validation

File: back/src/CiTron/User/Entity/User.php [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+        $this->roles->add($roles);
+        return $this;
+    }
+
+    /**
+     * @param Role $role
+     * @return self
+     */
+    /*
+    public function addRole(Role $role)
```
Comment (valanz) [inline back/src/CiTron/User/Entity/User.php]: Is it commented out on purpose?

File: back/app/config/routing/users.yml (hunk 3 above)
Comment (Nek-) [inline back/app/config/routing/users.yml]: :+1:

File: back/src/CiTron/Controller/User/RegistrationController.php (hunk 4 above)
Comment (Shine-neko) [inline back/src/CiTron/Controller/User/RegistrationController.php]: No interest here

File: back/src/CiTron/User/Form/UserType.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class UserType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('username')
```
Comment (Shine-neko) [inline back/src/CiTron/User/Form/UserType.php]: Deprecated.  Instead use 
 - Symfony\Component\Form\Extension\Core\Type\TextType
 - Symfony\Component\Form\Extension\Core\Type\PasswordType
- Symfony\Component\Form\Extension\Core\Type\EmailType

File: back/src/CiTron/Controller/User/RegistrationController.php (hunk 4 above)
Comment (valanz) [inline back/src/CiTron/Controller/User/RegistrationController.php]: @Shine-neko I wouldn't be so categorical.
Even if in this case the name is obvious, it's always nice to have some context about what the action is supposed to do.

File: back/src/CiTron/User/Form/UserType.php (hunk 7 above)
Comment (Nek-) [inline back/src/CiTron/User/Form/UserType.php]: Everything is API in this application. As the most difference between them is the template, we don't need to specify types.

File: back/src/CiTron/Controller/User/RegistrationController.php (hunk 4 above)
Comment (Nek-) [inline back/src/CiTron/Controller/User/RegistrationController.php]: @Shine-neko We understand the point of view of advanced programmers on comments. But this project has for goal to be open source and the most clear possible so eveybody can add its building block.

**Pull request merged**